lib/set-npmrc-auth.js

```diff
@@ -0,0 +1,26 @@
+const {appendFile} = require('fs-extra');
+const getAuthToken = require('registry-auth-token');
+const nerfDart = require('nerf-dart');
+const SemanticReleaseError = require('@semantic-release/error');
+const getRegistry = require('./get-registry');
+
+module.exports = async ({publishConfig, name}, logger) => {
+  const registry = await getRegistry(publishConfig, name);
+  logger.log('Verify authentication for registry %s', registry);
+  const {NPM_TOKEN, NPM_USERNAME, NPM_PASSWORD, NPM_EMAIL} = process.env;
+
+  if (!getAuthToken(registry)) {
+    if (NPM_USERNAME && NPM_PASSWORD && NPM_EMAIL) {
+      // Using the old auth token format is not considered part of the public API
+      // This might go away anytime (i.e. once we have a better testing strategy)
```

Comment:
can we drop support for `NPM_USERNAME` & `NPM_PASSWORD` and only support `NPM_TOKEN`?

Reply:
No unfortunately....It's because https://github.com/npm/npm-registry-couchapp doesn't accept login with a token. It accept only the legacy auth (username:password in base 64 + email).

I couldn't find a way to make https://github.com/npm/npm-registry-couchapp authentication works like https://www.npmjs.com/ with a token.